File: Videa/Programovací jazyky 2 z 2/Kvadratická rovnice.py
import math
print ("Program na výpočet kvadratické rovnice\nTvar kvadratické rovnice: ax^2+bx+c=0")
a=float(input("Zadej koeficient a: "))
b=float(input("Zadej koeficient b: "))
c=float(input("Zadej koeficient c: "))

#Výpočet diskriminantu
d=b*b-4*a*c

#Vyřešení rovnice
if d < 0:
    print ("Rovnice nemá řešení v oboru reálných čísel")
    input ("\nStiskněte enter pro konec programu...")
    exit()

if d == 0:
    x=-b/(2*a)
    print ("Rovnice má jedno řešení \nx1=", x)
    input ("\nStiskněte enter pro konec programu...")
    exit()

if d > 0:
    x1=(-b+math.sqrt(d))/(2*a)
    x2=(-b-math.sqrt(d))/(2*a)
    print ("Rovnice má 2 řešení:\nx1=", x1, "\nx2=", x2)
    input ("\nStiskněte enter pro konec programu...")
    exit()

    

File: Videa/Programovací jazyky 2 z 2/Kvadratická rovnice.c
#include <stdio.h>
#include <math.h>
#include <stdlib.h>

int main() {
	float a, b, c;
	printf("Program na vypocet kvadraticke rovnice\nTvar kvadraticke rovnice: ax^2+bx+c=0\n");
	printf("Zadej koeficient a: ");
	scanf_s("%f", &a);
	printf("Zadej koeficient b: ");
	scanf_s("%f", &b);
	printf("Zadej koeficient c: ");
	scanf_s("%f", &c);

	//Výpočet diskriminantu
	float d = pow(b, 2) - 4 * a * c;

	//Vyřešení rovnice
	if (d < 0) {
		printf("Rovnice nema reseni v oboru realnych cisel");
	}
	if (d == 0) {
		float x = -b / (2 * a);
		printf("Rovnice ma jedno reseni \nx = %f", x);
	}
	if (d > 0) {
		float x1 = (-b + sqrt(d)) / (2 * a);
		float x2 = (-b - sqrt(d)) / (2 * a);
		printf("Rovnice ma dve reseni \nx1 = %f \nx2 = %f", x1, x2);
	}
	printf("\n\nStisknete enter pro konec programu...");
	system("pause>nul");
	return 0;
}


